### ISSUE TITLE
Extend Interface: misleading error message

### ISSUE THREAD

**Issue by oliver-h** [Microsoft/TypeScript#9004]
**TypeScript Version:** 

1.8.10

**Code**

```ts
export interface MyInterface {
  foo(): void;
}

export class MyImplementation extends MyInterface {
  public foo(): void {
    console.log("foo!");
  }
}
```

**Expected behavior:**
Error TSXXXX: Cannot 'extend' interface. Use 'implements' instead.
**Actual behavior:** 
Error TS2304: Cannot find name 'MyInterface'.

**Comment by aluanhaddad** [comment]
Yeah, this message is very annoying.

**Comment by schultyy** [comment]
Is this up for grabs? If yes, I would like to fix this

**Comment by DanielRosenwasser** [comment]
Go for it @schultyy! Keep us posted!

**Comment by schultyy** [comment]
I started to dive into the codebase and probably found the place where the current error message is used: https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts#L7381
But I'm not sure completely sure where to make the change to use the new error instead. Can you give me a hint?

**Comment by weswigham** [comment]
Edit `diagnosticMessages.json` and run `jake generate-diagnostics` to regenerate the diagnostic message types.

**Comment by HerringtonDarkholme** [comment]
@schultyy I also have a try to add this new error report. You can take a look on it. I'm also new to tsc. So just take it as reference.
